odoo/fields.py

```diff
@@ -2531,6 +2531,7 @@ def __init__(self, selection=Default, string=Default, **kwargs):
     def setup_nonrelated(self, model):
         super().setup_nonrelated(model)
         assert self.selection is not None, "Field %s without selection" % self
+        assert (not isinstance(self.selection, list)) or all(map(lambda o: o[0], self.selection)), "Field %s has selection without key" % self
```

Comment:
`map` is useful when there is already a function which can be applied as-is to the input. Here it's pointless overhead, use a generator comprehension.

It is also unclear that this check, if it belongs at all, belongs here: the selection keys are typechecked in `_setup_attrs`. rco specifically moved that check from here to there in 1ca2c7e457a688b2288536416c0a7350172e0cda.

Reply:
Removed this: empty string is not an invalid value for the selection options from the perspective of the orm.